core/store/src/trie/iterator.rs

```diff
@@ -31,11 +31,14 @@ impl Crumb {
             self.status = CrumbStatus::Exiting;
             return;
         }
-        self.status = match (&self.status, &self.node.node) {
-            (_, &TrieNode::Empty) => CrumbStatus::Exiting,
+        if self.node.is_none() {
```

Comment:
`let Some(node) = &self.node else { ... return; }`

to avoid unwrap later. Same for all cases below.